src/client/producer_consumer_collection.go

```diff
@@ -64,23 +64,26 @@ func ConsumerWorker(consumerChan <-chan *args.JMXHost, wg *sync.WaitGroup, i *in
 
 		// Gather Metrics for consumer
 		if args.GlobalArgs.All() || args.GlobalArgs.Metrics {
-			// Lock since we can only make a single JMX connection at a time.
-			jmxwrapper.JMXLock.Lock()
 			log.Debug("Collecting metrics for consumer %s", consumerEntity.Metadata.Name)
 
-			// Open a JMX connection to the consumer
-			options := make([]jmx.Option, 0)
-			if args.GlobalArgs.KeyStore != "" && args.GlobalArgs.KeyStorePassword != "" && args.GlobalArgs.TrustStore != "" && args.GlobalArgs.TrustStorePassword != "" {
-				ssl := jmx.WithSSL(args.GlobalArgs.KeyStore, args.GlobalArgs.KeyStorePassword, args.GlobalArgs.TrustStore, args.GlobalArgs.TrustStorePassword)
-				options = append(options, ssl)
+			config := &gojmx.JMXConfig{
+				Hostname:         jmxInfo.Host,
+				Port:             int32(jmxInfo.Port),
+				Username:         jmxInfo.User,
+				Password:         jmxInfo.Password,
+				RequestTimeoutMs: int64(args.GlobalArgs.Timeout),
 			}
 
-			options = append(options, jmx.WithNrJmxTool(args.GlobalArgs.NrJmx))
+			if args.GlobalArgs.KeyStore != "" && args.GlobalArgs.KeyStorePassword != "" && args.GlobalArgs.TrustStore != "" && args.GlobalArgs.TrustStorePassword != "" {
+				config.KeyStore = args.GlobalArgs.KeyStore
+				config.KeyStorePassword = args.GlobalArgs.KeyStorePassword
+				config.TrustStore = args.GlobalArgs.TrustStore
+				config.TrustStorePassword = args.GlobalArgs.TrustStorePassword
+			}
 
-			if err := jmxwrapper.JMXOpen(jmxInfo.Host, strconv.Itoa(jmxInfo.Port), jmxInfo.User, jmxInfo.Password, options...); err != nil {
+			conn, err := jmxConnProvider.NewConnection(config)
+			if err != nil {
 				log.Error("Unable to make JMX connection for Consumer '%s': %s", consumerEntity.Metadata.Name, err.Error())
-				jmxwrapper.JMXClose() // Close needs to be called even on a failed open to clear out any set variables
-				jmxwrapper.JMXLock.Unlock()
 				continue
 			}
```

Comment:
I would also move this and in the producerWorker bellow Jmx bellow to a private getConsumerJMXConnection or ConsumerJMXConnect function, just to have it more readable.